BUILD

```diff
@@ -258,22 +258,60 @@ cc_binary(
 ################################################################################
 genrule(
     name = "generate_java_descriptor_proto",
-    tools = [":protoc"],
-    srcs = [ "src/google/protobuf/descriptor.proto", ],
-    outs = [ "com/google/protobuf/DescriptorProtos.java" ],
+    srcs = ["src/google/protobuf/descriptor.proto"],
+    outs = ["com/google/protobuf/DescriptorProtos.java"],
     cmd = "$(location :protoc) --java_out=$(@D)/../../.. $<",
+    tools = [":protoc"],
 )
 
 java_library(
     name = "java_proto",
-    visibility = ["//visibility:public"],
     srcs = glob([
-        "java/src/main/java/com/google/protobuf/*.java"
+        "java/src/main/java/com/google/protobuf/*.java",
     ]) + [
-      ":generate_java_descriptor_proto",
-    ]
+        ":generate_java_descriptor_proto",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+################################################################################
+# Python support
+################################################################################
+
+# A hack that moves Python runtime srcs files form python/google/protobuf/... to
+# bazel-genfiles/google/protobuf/... so that the generated files can correctly
+# import modules from google.protobuf.
+PYTHON_SRCS = glob(
+    ["python/google/protobuf/**/*.py"],
+    exclude = [
+        "python/google/protobuf/**/*_pb2.py",
+        "python/google/protobuf/internal/*_test.py",
+        "python/google/protobuf/internal/test_util.py",
+    ],
+)
+
+PYTHON_MOVED_SRCS = [x[len("python/"):] for x in PYTHON_SRCS]
+
+PYTHON_RUNTIME_SRCS = PYTHON_MOVED_SRCS + \
+                      [x[:-len(".proto")] + "_pb2.py" for x in WELL_KNOWN_PROTOS]
+
+genrule(
+    name = "generate_python_protos",
+    srcs = PYTHON_SRCS + ["src/" + x for x in WELL_KNOWN_PROTOS],
```

Comment:
If we have a py_library for the runtime files, then the import path of those modules will become:

python.google.protobuf.*

However, our generate code uses "from google.protobuf import Message".

Thus this hack to copy all the src files to the genfiles/google/protobuf directory. This is needed for clients which will import the whole protobuf package probably under //third_party/protobuf, in which case,  the module name will become "third_party.protobuf.python.google.protobuf.*". The generated file won't work. Copying the srcs to the genfiles is a hack to make this work.

Ideally py_library should contain an attribute to set the python path of the library, so the callsite could import the modules correctly.